src/lib/components/Accordion/AccordionItem.svelte

```diff
@@ -0,0 +1,52 @@
+<script lang="ts">
+  import { getContext, onMount } from "svelte";
+  import type { AccordionContext, AccordionContextItems } from "./types";
+
+  // Note: `id` has a fallback value, but ideally,
+  // one should supply their own identifiers
+  export let id = "accordionItem-" + Math.random().toString(36);
+  export let title: string | null | undefined = "Title";
+  export let expanded = false;
+
+  const context: AccordionContext = getContext("Accordion");
+
+  let ref: HTMLElement | null = null;
+  let unsubscribe: () => void | undefined;
+
+  onMount(() => {
+    return () => {
+      if (context) context.remove({ id, expanded: !expanded });
+      if (unsubscribe) unsubscribe();
+    };
+  });
+
+  $: if (context) {
+    context.add({ id, expanded });
+    unsubscribe = context.items.subscribe((value: AccordionContextItems) => {
```

Comment:
This reactive statement will rerun every time `expanded` changes, which means we will repeatedly subscribe to the context and throw away the old subscription. We should either call the old unsubscribe before reassigning it to the new subscription, or (preferably) refactor this so that the subscription doesn't depend on the value of `expanded`. Something like this should work (once other suggestions about the context methods are taken into account):

```ts
  const { items: accordionItems } = context;
  onMount(() => () => context && context.remove(id));
  $: if (context) context.set(id, expanded);
  $: if (context) expanded = $accordionItems[id] ?? false;
```

Using the auto-subscription like this should also allow us to omit the explicit `unsubscribe()` from the cleanup function returned by `onMount`.